lib/tokenize.cpp

```diff
@@ -9414,11 +9414,20 @@ void Tokenizer::simplifyCPPAttribute()
                 if (head && head->str() == "(" && isFunctionHead(head, "{|;")) {
                     head->previous()->isAttributeNodiscard(true);
                 }
-            } else if (Token::findsimplematch(tok->tokAt(2), "maybe_unused", tok->link())) {
+            } else if ((Token::findsimplematch(tok->tokAt(2), "maybe_unused", tok->link())
+                        && (mSettings.standards.c >= Standards::C23 || mSettings.standards.cpp >= Standards::CPP17))
```

Comment:
It feels like we should check `isC()` if we check `mSettings.standards.c`. And `isCPP()` for `mSettings.standards.cpp`

Reply:
Yes, we need to do that. Good catch.